### PR TITLE
Head fusion improvements

### PR DESCRIPTION
**Pull request by Robotgiggle**
- Lock head fusion recipes requiring certain ingredients (storm stone, nightdew, essences) behind obtaining those ingredients
- New head fusion recipes for phantom, strider, and piglin
- Head fusion guidebook entry now has two recipes per page

### CODE REVIEW

**Comment by Robotgiggle** [comment]
Lemme know if the piglin recipe is OK, I can edit or replace it if necessary

**Comment by DaFuqs** [comment]
Good job!
Your PR reminds me that we will have to add the modonomicon pages to the wiki

File: src/main/resources/data/spectrum/modonomicon/books/guidebook/entries/creating_life/creature_manipulation.json [hunk 1]
```diff
@@ -84,11 +86,6 @@
       "type": "spectrum:spirit_instiller_crafting",
       "recipe_id": "spectrum:spirit_instiller/head_fusion/variant_changing/shulker_pink",
       "text": "book.spectrum.guidebook.creature_manipulation.shulkers.text"
-    },
-    {
-      "type": "spectrum:spirit_instiller_crafting",
-      "recipe_id": "spectrum:spirit_instiller/head_fusion/variant_changing/fox_arctic",
-      "text": "book.spectrum.guidebook.creature_manipulation.foxes.text"
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/modonomicon/books/guidebook/entries/creating_life/creature_manipulation.json:91]: "book.spectrum.guidebook.creature_manipulation.foxes.text" is not used anymore and can be removed from the lang file

Comment (Robotgiggle) [reply src/main/resources/data/spectrum/modonomicon/books/guidebook/entries/creating_life/creature_manipulation.json:91]: ah whoops, forgot about that

**Pull request merged**